rust/nasl-builtin-network/src/socket.rs

```diff
@@ -0,0 +1,731 @@
+                ));
+            }
+        };
+
+        let port = get_kb_item(context, "Secret/kdc_port")?;
+
+        let port = match port {
+            NaslValue::Number(x) => {
+                if x <= 0 || x > 65535 {
+                    return Err(FunctionErrorKind::Diagnostic(
+                        "KB key 'Secret/kdc_port' out of range".to_string(),
+                        Some(port),
+                    ));
+                }
+                x
+            }
+            x => {
+                return Err(FunctionErrorKind::Diagnostic(
+                    "KB key 'Secret/kdc_port' is either missing or has wrong type".to_string(),
+                    Some(x),
+                ));
+            }
+        };
+        if port <= 0 || port > 65535 {
+            return Err(FunctionErrorKind::Diagnostic(
+                "KB key 'Secret/kdc_port' out of range".to_string(),
+                Some(NaslValue::Number(port)),
+            ));
+        }
+
+        let use_tcp: bool = get_kb_item(context, "Secret/kdc_use_tcp")?.into();
+
+        let socket = match use_tcp {
+            true => Self::open_tcp(&hostname, port as u16, None, None, None),
+            false => Self::open_udp(&hostname, port as u16),
+        }?;
+
+        let ret = self.add(socket);
+
+        Ok(NaslValue::Number(ret as i64))
+    }
+
+    /// Open a TCP socket to the target host.
+    /// This function is used to create a TCP connection to the target host.  It requires the port
+    /// number as its argument and has various optional named arguments to control encapsulation,
+    /// timeout and buffering.
+    /// It takes an unnamed integer argument (the port number) and four optional named arguments:
+    /// - bufsz: An integer with the the size buffer size.  Note that by default, no buffering is
+    ///   used.
+    /// - timeout: An integer with the timeout value in seconds.  The default timeout is controlled
+    ///   by a global value.
+    /// - transport: One of the ENCAPS_* constants to force a specific encapsulation mode or force
+    ///   trying of all modes (ENCAPS_AUTO). This is for example useful to select a specific TLS or
+    ///   SSL version or use specific TLS connection setup priorities.  See *get_port_transport for
+    ///   a description of the ENCAPS constants.
+    /// - priority A string value with priorities for an TLS encapsulation. For the syntax of the
+    ///   priority string see the GNUTLS manual. This argument is only used in ENCAPS_TLScustom
+    ///   encapsulation.
+    fn open_sock_tcp(
+        &self,
+        register: &Register,
+        context: &Context,
+    ) -> Result<NaslValue, FunctionErrorKind> {
+        // Get port
+        let port = get_pos_port(register)?;
+        let timeout = super::get_opt_int(register, "timeout");
+        let transport = super::get_opt_int(register, "transport").unwrap_or(-1);
+        let _priority = super::get_named_value(register, "priority")
```

Comment:
why is not used?

Reply:
I am currently not sure what to do with it. In the feed it is not used and also `rustls` does not support them. This is pure and deep GnuTLS stuff and is only used to define a custom TLS connection, that e.g. uses a different crypto provider. Not sure if we still should support it.

Reply:
ah, ok, wouldn't it better to comment it and just don't get it?